src/Generator/Passes/ParamTypeToInterfacePass.cs

```diff
@@ -7,20 +7,22 @@ public class ParamTypeToInterfacePass : TranslationUnitPass
     {
         public override bool VisitFunctionDecl(Function function)
         {
-            if (!function.IsOperator)
-            {
+            if (!function.IsOperator || function.Parameters.Count > 1)
                 ChangeToInterfaceType(function.ReturnType);
-                foreach (Parameter parameter in function.Parameters)
-                {
-                    ChangeToInterfaceType(parameter.QualifiedType);
-                }
-            }
+
+            if(function.OperatorKind != CXXOperatorKind.Conversion &&
+                                    function.OperatorKind != CXXOperatorKind.ExplicitConversion)
+                     foreach (var parameter in function.Parameters)
```

Comment:
I don't understand, why have you not added the check for OperatorParameter. If you had a test - please add one - it would've failed.